### PR TITLE
feat: add comment view (for workspace comments, and block comments for partners)

### LINKED ISSUE #7905: Implement basic CommentView class

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes #7905 

### Proposed Changes

Adds a comment view class which will later be used to 

### Test Coverage

Manually tested:
  * Typing into the block.
  * Setting the text programmatically.
  * Collapsing the block.
  * Collapsing the block programmatically.
  * Setting the block to be non-editable programatically (no way to do this manually).
  * Resizing the block.
  * Resizing the block programatically.
  * Moving the block programatically (no way to do this manually because that will be handled by whatever class is using the view).

### Documentation

There will need to be lots of docs later.

### Additional Information

Hopefully this is broken up in a way that's easy to review commit-wise 🤞 I do split some methods out into sub-procedures in later commits, but I think that's the only thing that majorly changes between earlier commits and later ones.

### CODE REVIEW

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [inline core/comments/comment_view.ts:110]: The disadvantage of being on the same layer as blocks is that a block can go over the comment and look like it's inside it. That's a bigger issue with mutators bubbles, since they always hold blocks. I'm neutral on which one we should do for comments.

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [inline core/comments/comment_view.ts:57]: Add note on coordinate system.

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [inline core/comments/comment_view.ts:16]: Check for TODOs and associate with issues before merging.

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [reply core/comments/comment_view.ts:373]: Same question for all listeners.

File: core/comments/comment_view.ts
Comment (BeksOmega) [reply core/comments/comment_view.ts:110]: There is also a weird thing where when we deserialize, the comments will be deserialized after blocks, so they'll always end up on top after load even if they were behind blocks before.

File: core/comments/comment_view.ts
Comment (BeksOmega) [reply core/comments/comment_view.ts:96]: Yesh! It is a little bit silly because if you try to reference the property in the creation method typescript doesn't realize you're doing a bad. (E.g. reference `this.textArea` in `createTextArea` throws a runtime error) But that's not specific to what I'm doing here, typescript just isn't smart enough to track references from the constructor.

File: core/comments/comment_view.ts
Comment (BeksOmega) [reply core/comments/comment_view.ts:573]: Can do, but I'll probably just give it the same format as blocks (`isDeadOrDying`) unless you have a different way you'd like it to look.

File: core/comments/comment_view.ts
Comment (BeksOmega) [inline core/comments/comment_view.ts:140]: Do you have thoughts on this @rachel-fenichel ?

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [reply core/comments/comment_view.ts:573]: That's fine. I just don't want to get to where we're inspecting whether a random SVG node is null as our way of checking for disposal >.<

Although that's no longer possible because of how you did initialization, so maybe it's fine to leave out `isDeadOrDying` and assume that we'll add it as soon as we have problems with events and disposal. Your call.

File: core/comments/comment_view.ts
Comment (BeksOmega) [reply core/comments/comment_view.ts:374]: - [ ] Add tests

File: core/comments/comment_view.ts
Comment (rachel-fenichel) [reply core/comments/comment_view.ts:146]: I don't know. Is replacing those images an immediate need, or a theoretical one?

File: tests/mocha/workspace_comment_test.js
Comment (rachel-fenichel) [inline tests/mocha/workspace_comment_test.js:12]: Unexpected suite.only. Please fix.

**Review by rachel-fenichel — APPROVED** [review]

File: core/comments/comment_view.ts
Comment (BeksOmega) [reply core/comments/comment_view.ts:143]: It's a need for spork blocks. For MakeCode we can just include their image in core.

Comment (BeksOmega) [reply core/comments/comment_view.ts:143]: @maribethb  as well.

Comment (maribethb) [reply core/comments/comment_view.ts:143]: Yeah, they do, but tbh they probably should anyway because the default one links to one hosted on app engine. I don't see this as being a concern, it's how the rest of media assets work. If we wanted to refactor that in the future that's a different discussion and shouldn't be folded into this work imo

Comment (BeksOmega) [reply core/comments/comment_view.ts:143]: Ok I'll dig into what MakeCode is currently doing and chat with them if it seems like it might be a problem. Since we've been holding v11 until we can have stuff ready for MakeCode, I want to make sure it actually is ready, and they're not going to have to wait for a v12.

Comment (maribethb) [reply core/comments/comment_view.ts:143]: What kind of problems are you thinking about? The media directory is already configurable via options, and they can copy the assets they want to keep the same into their own hosting solution (which is what we currently recommend, though not that loudly). I feel like I'm missing something here if you're worried about something needing a v12.

Comment (BeksOmega) [reply core/comments/comment_view.ts:143]: You're right I don't think this would need a v12. If anything I would just make it possible to configure the assets individually (which would be a feat not breaking). Based on some investigation it looks like they're already hosting everything themselves so it shouldn't be an issue. I just want to make sure we have everything pinned down so we can actually say MakeCode has been migrated.

**Pull request merged**
